C++/55_Jump_Game.cpp

```diff
@@ -0,0 +1,44 @@
+//55. Jump Game
+/*
+Given an array of non-negative integers, you are initially positioned at the first index of the array.
+
+Each element in the array represents your maximum jump length at that position.
+
+Determine if you are able to reach the last index.
+
+For example:
+A = [2,3,1,1,4], return true.
+
+A = [3,2,1,0,4], return false.
+
+Author: Xinyu Liu
+
+Tag: Array, Greedy
+*/
+
+#include <iostream>
+#include <vector>
+using namespace std;
+
+class Solution {
+public:
+    bool canJump(vector<int>& nums) {
+        int size_nums = nums.size();
+        int i = 0;
+        for (int reach = 0; i < size_nums && i <= reach; i++){
+            reach = max(nums.at(i) + i, reach);
+            if (reach >= size_nums -1)
+                return true;
+        }
+        return i == size_nums;
```

Comment:
i is only updated at the end of each loop, right? Suppose we are at the loop before i is size_nums, then `i = size_nums - 1`. `reach` is the maximum of `nums[i] + i` and `reach`. And `nums[i]` is non-negative, so `reach >= nums[i] + i >= i = size_nums - 1`, it will return true. Then i would never be `size_nums`.